astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -646,6 +647,32 @@ def test_read_complex():
         Table.read(txt, format='ascii.ecsv')
 
 
+@pytest.mark.skipif(NUMPY_LT_1_19,
+                    reason="numpy cannot parse 'complex' as string until 1.19+")
+def test_read_complex_v09():
```

Comment:
Is there something special about `complex` as opposed to `object`, `str`, or `datetime64`? AFAIK this test is exercising exactly the same code path as the one below so I'm not sure we need it, but maybe I'm missing something.

Reply:
There's nothing special other than it's one of only two tests that checks for the ValueError we disable. I thought there must be a reason for it existing so I made sure to test it versus the new changes, but I don't really see the reason either.

The test for complex will also break with the proposed API change to move from Exception to Warning, so it was on my mind.